riak_test/tests/sibling_benchmark.erl

```diff
@@ -0,0 +1,222 @@
+%%        %%{version, current}]
+%%      },
+
+
+-export([confirm/0]).
+-include_lib("eunit/include/eunit.hrl").
+
+%% keys for non-multipart objects
+-define(TEST_BUCKET,        "riak-test-bucket").
+-define(KEY_SINGLE_BLOCK,   "riak_test_key1").
+
+%% keys for multipart uploaded objects
+-define(KEY_MP,        "riak_test_mp").  % single part, single block
+
+
+
+confirm() ->
+    RTConfig = rt_config:get(sibling_benchmark, []),
+    Concurrency = proplists:get_value(write_concurrency, RTConfig, 2),
+
+    CSConfig0 = rtcs:replace_cs_config(connection_pools,
+                                       [
+                                        {request_pool, {Concurrency*2, 0} },
+                                        {bucket_list_pool, {rtcs:bucket_list_pool_size(), 0} }
+                                       ],
+                                       rtcs:cs_config()),
+    CSConfig1 = rtcs:replace_cs_config(leeway_seconds,
+                                       5, CSConfig0),
+    CSConfig = rtcs:replace_cs_config(gc_interval,
+                                       10, CSConfig1),
+    ConnConfig = [{cs, CSConfig}],
+
+    {UserConfig, {RiakNodes, _CSNodes, _Stanchion}} =
+        case proplists:get_value(version, RTConfig, current) of
+            current -> rtcs:setup(4, ConnConfig);
+            previous ->
+                PrevConfig =  ConnConfig ++ rtcs:previous_configs(),
+                rtcs:setup(4, PrevConfig, previous)
+        end,
+
+    %% setting up the stage
+    ?assertEqual([{buckets, []}], erlcloud_s3:list_buckets(UserConfig)),
+    lager:info("creating bucket ~p", [?TEST_BUCKET]),
+    ?assertEqual(ok, erlcloud_s3:create_bucket(?TEST_BUCKET, UserConfig)),
+    %% The first object
+    erlcloud_s3:put_object(?TEST_BUCKET, ?KEY_SINGLE_BLOCK, <<"boom!">>, UserConfig),
+
+    get_counts(RiakNodes, ?TEST_BUCKET, ?KEY_SINGLE_BLOCK),
+    lager:info("====================== run benchmark ====================="),
+    {ok, Pid} = start_stats_checker(RiakNodes),
+
+    lager:info("write_concurrency: ~p", [Concurrency]),
+    Writers = [start_object_writer(UserConfig, 100) || _ <- lists:seq(1, Concurrency)],
+    {ok, Reader} = start_object_reader(UserConfig),
+
+    %% Do your madness or evil here
+    Duration = proplists:get_value(duration_sec, RTConfig, 16),
+
+    timer:sleep(Duration * 1000),
+    LeaveAndJoin = proplists:get_value(leave_and_join, RTConfig, 0),
+    leave_and_join_node(RiakNodes, LeaveAndJoin),
+
+    ok = stop_object_reader(Reader),
+    [stop_object_writer(Writer) || {ok, Writer} <- Writers],
+    lager:info("====================== benchmark done ===================="),
+    {ok, MaxSib} = stop_stats_checker(Pid),
+    %% Max number of siblings should not exceed number of upload concurrency
+    %% according to DVVset implementation
+    lager:info("MaxSib:Concurrency = ~p:~p", [MaxSib, Concurrency]),
+    %% Real concurrency is, added by GC workers
+    ?assert(MaxSib =< Concurrency + 5),
```

Comment:
For asserting the validity of whole test process, using max siblings count during test processes may be better.
But using the latest value is not bad because siblings count usually increase (almost) monotonically or explode instantaniously :sweat_smile: